tests/rptest/scale_tests/upgrade_availability_test.py

```diff
@@ -0,0 +1,293 @@
+            timeout_sec=60,
+            err_msg='Producer stopped writing')
+
+        # Wait until all replicas have the same data
+        self.admin.wait_stable_configuration(topic=self.topic,
+                                             namespace='kafka',
+                                             replication=3,
+                                             timeout_s=120)
+
+        # Wait until movement status is done on all nodes
+        self.confirm_partition_movement_status(self.rp_hostnames,
+                                               'done',
+                                               timeout_s=300)
+
+        # Terminate when the producer finishes
+        self.producer.wait(timeout_sec=1200)
+
+    def run_partition_movement_for_n_seconds(self,
+                                             hosts: list[str],
+                                             timeout_s: int = 10):
+        def check_for_done_status_on_replicas():
+            replicas_status = []
+
+            for host in hosts:
+                res = self.admin._get_configuration(host,
+                                                    namespace='kafka',
+                                                    topic=self.topic,
+                                                    partition=0)
+
+                if 'status' in res:
+                    status = res['status']
+                    replicas_status.append(status == 'done')
+
+            if any(replicas_status):
+                raise Exception('partition movement finished on a replica')
+
+            return False
+
+        try:
+            wait_until(check_for_done_status_on_replicas,
+                       timeout_sec=timeout_s)
+        except TimeoutError as ex:
+            # Ignore the timeout. We want that to happen
+            pass
+
+    @cluster(num_nodes=5, log_allow_list=CHAOS_LOG_ALLOW_LIST)
+    def test_node_will_not_be_in_replica_set(self):
+        rs_tuple = self.run_workload_and_get_replica_sets()
+        nodes_in_replica_set = rs_tuple[0]
+        nodes_not_in_replica_set = rs_tuple[1]
+
+        # Execute partition movement such that node X will no longer be in the
+        # replica set. Therefore, move the partition to a node that is not in
+        # the replica set.
+        node_x = choice(nodes_in_replica_set)
+        node_y = choice(nodes_not_in_replica_set)
+        print(f'Move partition from {node_x} to {node_y}')
+        part_movement_th = threading.Thread(
+            target=self.move_from_src_to_dest,
+            args=(node_x, node_y),
+        )
+        part_movement_th.start()
+
+        # Check that partition movement started/is on-going
+        self.confirm_partition_movement_status(self.rp_hostnames,
+                                               'in_progress')
+
+        self.run_partition_movement_for_n_seconds(self.rp_hostnames,
+                                                  timeout_s=10)
```

Comment:
It looks like 10 seconds is too long as some replicas are already finishing partition movement within that time. I'll need to figure out the sweet spot for the timeout.